rs/orchestrator/src/catch_up_package_provider.rs

```diff
@@ -210,14 +223,78 @@ impl CatchUpPackageProvider {
     // caller.
     async fn fetch_catch_up_package(
         &self,
-        url: Url,
+        node_id: &NodeId,
+        url: String,
         param: Option<CatchUpPackageParam>,
     ) -> Option<pb::CatchUpPackage> {
-        Agent::new_with_client(self.client.clone(), url, Sender::Anonymous)
-            .query_cup_endpoint(param)
+        let body = Bytes::from(
+            param
+                .and_then(|param| serde_cbor::to_vec(&param).ok())
+                .unwrap_or_default(),
+        );
+
+        let client_config = self
+            .crypto_tls_config
+            .client_config(*node_id, self.registry.get_latest_version())
+            .map_err(|e| warn!(self.logger, "Failed to create tls client config: {:?}", e))
+            .ok()?;
+
+        let https = HttpsConnectorBuilder::new()
+            .with_tls_config(client_config)
+            .https_only()
+            .enable_all_versions()
+            .build();
+
+        let client = Client::builder(TokioExecutor::new())
+            .pool_idle_timeout(tokio::time::Duration::from_secs(600))
```

Comment:
Is there a reason for this value?

Reply:
It is the [default](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/canister_client/src/http_client.rs?L45) value of the http client config which was [used](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/orchestrator/src/catch_up_package_provider.rs?L82) before my changes